File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-212215-18237/analysis/own-tests-project/src/surveycrawler/crawler_run.py
class CrawlerRun:

    # undoing

    def register_undo(self, an_undo_action):
        raise NotImplementedError()

    # guarded run

    def start_guarded_run(self, a_crawler):
        raise NotImplementedError()

    def end_guarded_run(self, a_crawler):
        raise NotImplementedError()

    def finish(self, a_crawler):
        raise NotImplementedError()

    # error handling

    def handle_error(self, a_crawler):
        raise NotImplementedError()

    # moving

    def slide_crawler_to(self, a_crawler, a_position):
        raise NotImplementedError()

File: experiments/020-PythonVsJava/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260921-204934-85557/output/project/src/main/java/surveycrawler/CrawlerRun.java
package surveycrawler;

public abstract class CrawlerRun {

    // moving

    public abstract Point positionAfterMovingOn(SeabedGround aGround, Point aPosition, Point aDirection);

    public abstract void register(CrawlerMovement aMovement);

    // guarded run

    public abstract CrawlerRun startGuardedRun();

    public abstract CrawlerRun endGuardedRunOn(SurveyCrawler aCrawler);

    public abstract void assertIsFinished();

    // undoing

    public abstract void undoOn(SurveyCrawler aCrawler);
}
